dspace-server-webapp/src/main/java/org/dspace/app/rest/BundleUploadBitstreamController.java

```diff
@@ -111,6 +124,30 @@ public ResponseEntity<RepresentationModel<?>> uploadBitstream(
             throw new UnprocessableEntityException("The InputStream from the file couldn't be read", e);
         }
 
+        // We do this before calling `updateBitstream` because that function calls `context.commit`
+        // Add suitable provenance
+        EPerson e = context.getCurrentUser();
+        String timestamp = DCDate.getCurrent().toString();
+        Item item = bundle.getItems().get(0);
+
+        StringBuilder prov = new StringBuilder();
+        prov.append("Item was added bitstream to bundle (").append(bundle.getID())
+                .append(") by ").append(e.getFullName()).append(" (").append(e.getEmail())
+                .append(") on ").append(timestamp).append("\n");
+        try {
+            prov.append(installItemService.getBitstreamProvenanceMessage(context, item));
+            itemService.addMetadata(context, item, MetadataSchemaEnum.DC.getName(),
+                    "description", "provenance", "en", prov.toString());
+            //Update item in DB
+            itemService.update(context, item);
+        } catch (SQLException ex) {
+            throw new RuntimeException("SQLException in BundleUploadBitstreamConverter.uploadBitstream when " +
```

Comment:
Are you sure we want to throw RuntimeException when we catch SQLException? Please consider throwing SQLException with some custom message instead of RuntimeEx...

Reply:
But if I used SQLException I would have to add it to method signature what I think we don't want because that is not our method or?+